lua/entities/gmod_wire_expression2/base/parser.lua

```diff
@@ -90,6 +94,23 @@ function Parser.Execute(...)
 	return xpcall(Parser.Process, E2Lib.errorHandler, instance, ...)
 end
 
+function Parser.DumpTree(tree, indentation)
+	indentation = indentation or ''
+	local str = indentation .. tree[1] .. '\n'
```

Comment:
It would be easy to print the trace (in `tree[2]`) next to the instruction type.